.test-infra/metrics/sync/github/sync_workflows.py

```diff
@@ -96,64 +115,80 @@ def getToken():
 def fetchWorkflowData():
     '''Return a json with all the workflows and the latests
     ten executions'''
-    listOfWorkflows = {}
-    workflowsStatus = {}
+    completed = False
+    page = 1 
+    workflows = []
+    retryFactor = 1
     try:
-        url = "https://api.github.com/repos/apache/beam/actions/workflows"
-        queryOptions = { 'branch' : 'master', 'per_page' : GH_NUMBER_OF_WORKFLOWS }
-        response = requests.get(url = url, params = queryOptions)
-        jsonResponse = response.json()
-        workflows = jsonResponse['workflows']
-        for item in workflows:
-            path =(item['path'])
-            isPostCommit = re.search('(.*)postcommit(.*)',path)
-            if isPostCommit:
-                result = re.search('/(.*).yml', path)
-                path =(result.group(1)) + ".yml"
-                GH_WORKFLOWS_YML_FILENAME.append(path)
-                listOfWorkflows[(item['id'])] = item['name']
-        url = "https://api.github.com/repos/apache/beam/actions/workflows/"
-        queryOptions = { 'branch' : 'master', 'per_page' : GH_WORKFLOWS_NUMBER_EXECUTIONS,
-                    'page' :'1', 'exclude_pull_request':True }
-        #headers = {'Authorization': 'Bearer {}'.format(getToken())}
-        for key in listOfWorkflows:
-            response = requests.get(url = "{}{}/runs".format(url,key),
+        while not completed:
+            url = "https://api.github.com/repos/apache/beam/actions/workflows"
+            queryOptions = { 'branch' : 'master', 'page': page, 'per_page' : GH_NUMBER_OF_WORKFLOWS }
+            response = requests.get(url = url, params = queryOptions)
+            jsonResponse = response.json()
+            if jsonResponse['total_count'] >= 100:
+                page = page + 1
+                workflowsPage = jsonResponse['workflows']
+                workflows.append(workflowsPage)
+            else:
+                completed = True
+                workflowsPage = jsonResponse['workflows']
+                workflows.append(workflowsPage)
+        for pageItem in workflows:
+            for item in pageItem:
+                path =item['path']
+                isPostCommit = re.search('(.*)postcommit(.*)',path)
+                if isPostCommit:
+                    result = re.search('/(.*).yml', path)
+                    path =(result.group(1)) + ".yml"
+                    workflowObject = Workflow(item['id'],item['name'],path)
+                    WORKFLOWS_OBJECT_LIST.append(workflowObject)
+            url = "https://api.github.com/repos/apache/beam/actions/workflows/"
+            queryOptions = { 'branch' : 'master', 'per_page' : GH_WORKFLOWS_NUMBER_EXECUTIONS,
+                        'page' :'1', 'exclude_pull_request':True }
+            #headers = {'Authorization': 'Bearer {}'.format(getToken())}
+        for workflow in WORKFLOWS_OBJECT_LIST:
+            requestSucceeded = False
+            while not requestSucceeded:
```

Comment:
We should also add a retry limit - we should give up if we've been retrying for longer than an hour and receive another error code so that we don't spin here forever.